ycmd/tests/get_completions_test.py

```diff
@@ -115,9 +123,800 @@ def GetCompletions_IdentifierCompleter_WorksForSpecialIdentifierChars_test():
                            completion_data ).json[ 'completions' ]
 
   assert_that( results,
-               has_items( CompletionEntryMatcher( 'font-size' ),
-                          CompletionEntryMatcher( 'font-family' ) ) )
+               has_items( CompletionEntryMatcher( 'font-size', '[ID]' ),
+                          CompletionEntryMatcher( 'font-family', '[ID]' ) ) )
+
 
+@with_setup( Setup )
+def GetCompletions_Fall_Back_ClangCompleter_test():
+
+  # because we use it a lot
+  NO_COMPLETIONS_ERROR = ErrorMatcher( RuntimeError, NO_COMPLETIONS_MESSAGE )
+
+  """
+  GetCompletions_Fall_Back_test
+
+  Tests for:
+    - [X] multiple languages: (use cases)
+      - [X] C:      macros
+      - [X] C++:    templates
+      - [X] Python: properties set from external method
+    - [X] Forced vs. Triggered semantic completion
+    - [X] Identifiers tagged with '[ID]'
+    - [X] Filter/sort based on query
+    - [X] Min chars for completion (only applies to ID)
+    - [X] Extra conf throws exception
```

Comment:
Interesting perspective. Happy to change this.

FWIW my perspective has always been that it's really useful to write regression tests (in particular integration tests testing an interface - as these are) from the functional perspective, so that if the code is changed (arbitrarily) then the tests should still work. I've always extended that to mean "even if the underlying codepath is similar, then the functional test still stands". i.e. if we made a change to the python completer which somehow interfaced badly with this, then we'd catch it.

Like I said, happy to change this.